homeassistant/components/weatherkit/config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Adds config flow for WeatherKit."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from apple_weatherkit.client import (
+    WeatherKitApiClient,
+    WeatherKitApiClientAuthenticationError,
+    WeatherKitApiClientCommunicationError,
+    WeatherKitApiClientError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    LocationSelector,
+    LocationSelectorConfig,
+    TextSelector,
+    TextSelectorConfig,
+)
+
+from .const import (
+    CONF_KEY_ID,
+    CONF_KEY_PEM,
+    CONF_SERVICE_ID,
+    CONF_TEAM_ID,
+    DOMAIN,
+    LOGGER,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Home"): str,
```

Comment:
We want to avoid having people set names themselves in new integrations/config flows and are slowly removing it in current integrations as well. People can set the name by renaming the device and or config entry.